subsys/bluetooth/audio/has.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+#include <device.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/audio/audio.h>
+#include <bluetooth/audio/capabilities.h>
+
+#include "has_internal.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HAS)
+#define LOG_MODULE_NAME bt_has
+#include "common/log.h"
+
+static struct bt_has has;
+
+static ssize_t read_features(struct bt_conn *conn, const struct bt_gatt_attr *attr, void *buf,
+			     uint16_t len, uint16_t offset)
+{
+	BT_DBG("conn %p attr %p offset %d", conn, attr, offset);
+
+	if (offset > sizeof(has.features)) {
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_OFFSET);
+	}
+
+	return bt_gatt_attr_read(conn, attr, buf, len, offset, &has.features,
+				 sizeof(has.features));
+}
+
+/* Hearing Access Service GATT Attributes */
+BT_GATT_SERVICE_DEFINE(has_svc,
+	BT_GATT_PRIMARY_SERVICE(BT_UUID_HAS),
+	BT_GATT_CHARACTERISTIC(BT_UUID_HAS_HEARING_AID_FEATURES, BT_GATT_CHRC_READ,
+			       BT_GATT_PERM_READ_ENCRYPT, read_features, NULL, NULL),
+);
+
+static int has_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/* Initialize the supported features characteristic value */
+	has.features = CONFIG_BT_HAS_HEARING_AID_TYPE;
+
+	if (IS_ENABLED(CONFIG_BT_HAS_HEARING_AID_BANDED)) {
+		/* HAP_d1.0r00; 3.7 BAP Unicast Server role requirements
+		 * A Banded Hearing Aid in the HA role shall set the Front Left and the Front
+		 * Right bits to a value of 0b1 in the Sink Audio Locations characteristic value.
+		 */
+		bt_audio_capability_set_location(BT_AUDIO_SINK, BT_AUDIO_LOCATION_FRONT_LEFT |
+								BT_AUDIO_LOCATION_FRONT_RIGHT);
+	} else if (IS_ENABLED(CONFIG_BT_HAS_HEARING_AID_BINAURAL_LEFT)) {
+		/* HAP_d1.0r00; 3.7 BAP Unicast Server role requirements
+		 * If the HA is part of a Binaural Hearing Aid Set, the HA shall set at least one
+		 * of the Front Left bit or the Front Right bit to a value of 0b1 in the Sink
+		 * Audio Locations characteristic value as defined in [2].
+		 */
+		bt_audio_capability_set_location(BT_AUDIO_SINK, BT_AUDIO_LOCATION_FRONT_LEFT);
+	} else {
+		/* Monaural and Binaural Right */
+		bt_audio_capability_set_location(BT_AUDIO_SINK, BT_AUDIO_LOCATION_FRONT_RIGHT);
```

Comment:
Would `BT_AUDIO_LOCATION_FRONT_CENTER` make more sense for mono?

Reply:
Unsure about this.  If the hearing aid is supposed to receive monaural sound, it should not set a location at all.  ("Mono" equals "no location".)  Or, if it supposed to receive audio for one ear, it may be either FRONT_LEFT or FRONT_RIGHT, depending upon which ear it is placed on.
I will ask on the reflector.

Reply:
(FRONT_CENTER is almost certainly not the correct thing to set.)

Reply:
The Location of 0 is Not Allowed as defined in Generic Audio Assigned Numbers. I recall such discussion, the conclusion was that there shall be location set.

![image](https://user-images.githubusercontent.com/10126877/158381436-3f61d181-b585-4fad-b181-6ec1a96d2f41.png)

Reply:
(That may lead us to larger changes, so OK to leave as is for now.)

Reply:
To me it sounds like mono hearing assistant should not set location, and that the location characteristic should not be there. 

This is not yet possible, so created https://github.com/zephyrproject-rtos/zephyr/issues/43851. 

Please add a TODO for it here, otherwise this PR LGTM

Reply:
The response from Chris also suggests that a mono HA may set either left or right, depending on what ear the device fits in. 

In that case, we could/should use `BT_HAS_HEARING_AID_MONAURAL_LEFT` or `BT_HAS_HEARING_AID_MONAURAL_RIGHT`. 

Alternatively, decouple the location from the `BT_HAS_HEARING_AID_` type, and add a separate Kconfig